src/main/java/edu/cornell/kfs/pdp/businessobject/lookup/CuACHPayeeLookupableHelperServiceImpl.java

```diff
@@ -31,8 +47,99 @@
 
         // add entity
         personFieldValues.put(KIMPropertyConstants.Person.ENTITY_ID, fieldValues.get(KIMPropertyConstants.Person.ENTITY_ID));
+        // add principal name
+        personFieldValues.put(KIMPropertyConstants.Person.PRINCIPAL_NAME, fieldValues.get(KIMPropertyConstants.Person.PRINCIPAL_NAME));
 
         return personFieldValues;
     }
 
+    /**
+     * Overridden to only search for employee or entity payees when principal name is specified.
+     * 
+     * @see org.kuali.kfs.pdp.businessobject.lookup.ACHPayeeLookupableHelperServiceImpl#getSearchResults(java.util.Map)
+     */
+    @SuppressWarnings({"deprecation", "rawtypes", "unchecked"})
```

Comment:
This method is just a tweaked copy of a superclass method, and the superclass method contains code that brings up these warnings.